feat: add 'download' attribute to boolean attributes list

cstmth opened the pull request.

fabianmichael commented:
@cstmth Thanks for your suggestion. Unfortunately, the download attribute is not a pure boolean attribute, but can also have a value. Would you please add a comment and a unit test, to ensure that it works wird different values as expected? Then I am hapyp to mrge your PR. Thanks :-)

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/a

cstmth commented:
Hey @fabianmichael, thanks for the quick response!

- I added a test verifying that an attribute in the `booleanAttributes` array can still have an additional value supplied.
- Also I added an explanatory comment to the entry in the `booleanAttributes` array.
- Furthermore, I found minor spelling mistakes that I also fixed.

Thanks for this great plugin!

fabianmichael commented:
@cstmth Thanks :-)
